File: app/src/main/java/ru/mpei/vmss/myapplication/fragments/Articles.java
package ru.mpei.vmss.myapplication.fragments;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.JsonArrayRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import ru.mpei.vmss.myapplication.R;
import ru.mpei.vmss.myapplication.activities.ArticleActivity;
import ru.mpei.vmss.myapplication.adapters.ArticleAdapter;
import ru.mpei.vmss.myapplication.pojo.Article;

public class Articles extends Fragment {

    private Context context;
    private ArticleAdapter adapter = null;
    private List<Article> dataList = new ArrayList<>();
    private String url;
    private String header;
    private String prefix;
    private int type;

    public Articles(){}

    public Articles(int type) {
        this.type = type;
    }

    @SuppressLint("NewApi")
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        context = Objects.requireNonNull(getActivity()).getApplicationContext();

        if(type == 0){
            this.url = context.getString(R.string.dashboardUrl);
            this.header = context.getString(R.string.dashboard);
            this.prefix = context.getString(R.string.imagesDashboardUrl);
        } else if(type == 1){
            this.url = context.getString(R.string.newsUrl);
            this.header = context.getString(R.string.news);
            this.prefix = context.getString(R.string.imagesNewsUrl);
        }

        View rootView = inflater.inflate(R.layout.fragment_articles, container, false);

        TextView title = rootView.findViewById(R.id.articlesHeader);
        title.setText(header);

        RecyclerView list = rootView.findViewById(R.id.articlesList);

        LinearLayoutManager layoutManager = new LinearLayoutManager(context);
        list.setLayoutManager(layoutManager);

        adapter = new ArticleAdapter(getActivity().getApplicationContext(), dataList, article -> {
            Intent intent = new Intent(context, ArticleActivity.class);
            intent.putExtra("head", article.getTitle());
            intent.putExtra("date", article.getDate());
            intent.putExtra("content", article.getContent());
            intent.putExtra("imageUrl", article.getImageUrl());
            startActivity(intent);
        });
        list.setAdapter(adapter);
        updateList();

        SwipeRefreshLayout refresher = rootView.findViewById(R.id.articlesRefresher);
        refresher.setColorSchemeColors(context.getColor(R.color.bgBottomNavigation));
        refresher.setOnRefreshListener(() -> {
            updateList();
            refresher.setRefreshing(false);
        });

        return rootView;
    }

    private void updateList(){
        dataList.clear();

        JsonArrayRequest request = new JsonArrayRequest(Request.Method.GET,
                this.url, response -> {
            for (int i = 0; i < response.length(); i++){
                JSONObject obj;
                try {
                    obj = response.getJSONObject(i);
                    dataList.add(new Article(obj.getString("name"),
                            obj.getString("content"),
                            obj.getString("chislo") + " " + obj.getString("month") + " в " + obj.getString("hour"),
                            null,
                            prefix+obj.getString("image_src")));
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
            adapter.setArticles(dataList);
        },
                error -> Toast.makeText(context, "Connection error", Toast.LENGTH_LONG).show());

        RequestQueue requestQueue = Volley.newRequestQueue(context);
        requestQueue.add(request);
    }
}

File: app/src/main/java/ru/mpei/vmss/myapplication/fragments/Articles.kt
package ru.mpei.vmss.myapplication.fragments

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.VolleyError
import com.android.volley.toolbox.JsonArrayRequest
import com.android.volley.toolbox.Volley
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import ru.mpei.vmss.myapplication.R
import ru.mpei.vmss.myapplication.activities.ArticleActivity
import ru.mpei.vmss.myapplication.adapters.ArticleAdapter
import ru.mpei.vmss.myapplication.adapters.ArticleAdapter.OnArticleClickListener
import ru.mpei.vmss.myapplication.pojo.Article
import kotlinx.android.synthetic.main.fragment_articles.*
import java.util.*

class Articles : Fragment {
    private var adapter: ArticleAdapter? = null
    private val dataList: MutableList<Article> = ArrayList()
    private var url: String? = null
    private var header: String? = null
    private var prefix: String? = null
    private var type = 0

    constructor()
    constructor(type: Int) {
        this.type = type
    }

    @SuppressLint("NewApi")
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        if (type == 0) {
            url = requireContext().getString(R.string.dashboardUrl)
            header = requireContext().getString(R.string.dashboard)
            prefix = requireContext().getString(R.string.imagesDashboardUrl)
        } else if (type == 1) {
            url = requireContext().getString(R.string.newsUrl)
            header = requireContext().getString(R.string.news)
            prefix = requireContext().getString(R.string.imagesNewsUrl)
        }
        articlesHeader.text = header
        val layoutManager = LinearLayoutManager(context)
        articlesList.layoutManager = layoutManager
        adapter = ArticleAdapter(activity!!.applicationContext, dataList, object : OnArticleClickListener {
            override fun onClickListener(article: Article?) {
                val intent = Intent(context, ArticleActivity::class.java)
                if (article != null) {
                    intent.putExtra("head", article.title)
                    intent.putExtra("date", article.date)
                    intent.putExtra("content", article.content)
                    intent.putExtra("imageUrl", article.imageUrl)
                }
                startActivity(intent)
            }
        })
        articlesList.adapter = adapter
        updateList()
        articlesRefresher.setColorSchemeColors(requireContext().getColor(R.color.bgBottomNavigation))
        articlesRefresher.setOnRefreshListener {
            updateList()
            articlesRefresher.isRefreshing = false
        }
        return inflater.inflate(R.layout.fragment_articles, container, false)
    }

    private fun updateList() {
        dataList.clear()
        val request = JsonArrayRequest(Request.Method.GET,
                url, Response.Listener { response: JSONArray ->
            for (i in 0 until response.length()) {
                var obj: JSONObject
                try {
                    obj = response.getJSONObject(i)
                    dataList.add(Article(obj.getString("name"),
                            obj.getString("content"),
                            obj.getString("chislo") + " " + obj.getString("month") + " в " + obj.getString("hour"),
                            null,
                            prefix + obj.getString("image_src")))
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            }
            adapter!!.setArticles(dataList)
        },
                Response.ErrorListener { _: VolleyError? -> Toast.makeText(context, "Connection error", Toast.LENGTH_LONG).show() })
        val requestQueue = Volley.newRequestQueue(context)
        requestQueue.add(request)
    }
}